aeppl/joint_logprob.py

```diff
@@ -175,33 +175,39 @@ def factorized_joint_logprob(
             for node in io_toposort(graph_inputs(q_logprob_vars), q_logprob_vars):
                 compute_test_value(node)
 
-    missing_value_terms = set(original_values.values()) - set(logprob_vars.keys())
+    original_values_ordered = tuple(rv_values.values())
+
+    missing_value_terms = set(original_values_ordered) - set(logprob_vars.keys())
     if missing_value_terms:
         raise RuntimeError(
             f"The logprob terms of the following value variables could not be derived: {missing_value_terms}"
         )
 
-    return logprob_vars
+    # Returned logprob terms sorted by order of input `rv_values`
+    return {
+        original_value: logprob_vars[original_value]
+        for original_value in original_values_ordered
+    }
```

Comment:
This can instead be done in `joint_logprob`, where it actually matters that the order of the returned graphs be predictable (if we decide to keep both functions)